File: Backend/csharp/NoteS/models/entity/Tag.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NoteS.models.entity;

[Table("tags")]
public partial class Tag(string name)
{
    [Column("id")] public int? Id { get; init; }

    [Column("name")] [MaxLength(64)] public string Name { get; init; } = name;

    [Column("account_id")] public int? Owner { get; set; }
    [Column("color")] [Range(0, 0xFFFFFF)] public int Color { get; set; } = 0;
}

File: Backend/java/src/main/java/com/notes/models/entity/Tag.java
package com.notes.models.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;
import org.hibernate.validator.constraints.Range;

import static jakarta.persistence.GenerationType.IDENTITY;
import static lombok.AccessLevel.PROTECTED;

@Entity
@Table(name = "tags")
@NoArgsConstructor(access = PROTECTED)
@AllArgsConstructor(access = PROTECTED)
@Builder
@Getter
public class Tag {
   @Id
   @GeneratedValue(strategy = IDENTITY)
   Long id;

   @NotNull
   @NotEmpty
   @Length(max = 128)
   @Column(name = "name", nullable = false, unique = true)
   String name;

   @NotNull
   @ManyToOne(fetch = FetchType.LAZY, optional = false)
   @JoinColumn(name = "account_id", nullable = false)
   Account owner;

   @NotNull
   @Range(min = 0, max = 0xFFFFFF)
   @Column(name = "color", nullable = false)
   Integer color;
}